### PR TITLE
For #25145: Unify behavior for opening systems setting for updating the default browser preference

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Mugurell** [comment]
This is great to reduce duplicated code. And add support for the newer APIs.
But there seems like some tests are failing:

>   TEST: process make_default_browser deep link for API 23 and below
  TEST: process make_default_browser deep link for API 23 and below[22]
  TEST: process make_default_browser deep link for API 23 and below[21]
  
all with the same cause:
```
  argument: FromSettings, matcher: eq(FromGlobal), result: -
  argument: mozilla.components.concept.engine.EngineSession$LoadUrlFlags@0, matcher: eq(mozilla.components.concept.engine.EngineSession$LoadUrlFlags@4), result: -
```
Maybe `openSetDefaultBrowserOption` needs to support this two arguments being passed in?

**Comment by Amejia481** [comment]
> This is great to reduce duplicated code. And add support for the newer APIs. But there seems like some tests are failing:
> 
> > TEST: process make_default_browser deep link for API 23 and below
> > TEST: process make_default_browser deep link for API 23 and below[22]
> > TEST: process make_default_browser deep link for API 23 and below[21]
> 
> all with the same cause:
> 
> ```
>   argument: FromSettings, matcher: eq(FromGlobal), result: -
>   argument: mozilla.components.concept.engine.EngineSession$LoadUrlFlags@0, matcher: eq(mozilla.components.concept.engine.EngineSession$LoadUrlFlags@4), result: -
> ```
> 
> Maybe `openSetDefaultBrowserOption` needs to support this two arguments being passed in?

Thanks for checking, I'll update the tests :)

**Comment by Amejia481** [comment]
I updated the PR :)

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt [hunk 1]
```diff
@@ -64,8 +65,14 @@ fun Activity.breadcrumb(
 
 /**
  * Opens Android's Manage Default Apps Settings if possible.
+ * Otherwise navigates to the Sumo article indicating why it couldn't open it.
+ * @param from fallback direction in case, couldn't open the setting.
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/ext/Activity.kt:69]: nit: In most places we have a `newline` in between description and params.

**Review by Mugurell — APPROVED** [review]
Looks good to me.
Seems CI also agrees.

File: app/src/main/java/org/mozilla/fenix/ext/Activity.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/ext/Activity.kt:69]: 👍🏽

**Pull request merged**